### ISSUE TITLE
Pulse simulator fails from pypi packages

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-aer#602]
### Informations

- **Qiskit Aer version**: 0.4.0
- **Python version**: 3.5, 3.6, 3.7, 3.8
- **Operating system**: Linux

### What is the current behavior?
Running the verify_wheels script from master (just to get #584) with the wheel from pypi results in errors and it doesn't run:

```
/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/__init__.py:60: RuntimeWarning: Could not import the IBMQ provider from the qiskit-ibmq-provider package. Install qiskit-ibmq-provider or check your installation.
  warnings.warn('Could not import the IBMQ provider from the '
/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/models/pulsedefaults.py:166: UserWarning: `qubit_freq_est` and `meas_freq_est` now have units of Hertz(Hz) rather than gigahertz(GHz).
  warnings.warn('`qubit_freq_est` and `meas_freq_est` now have units of '
RuntimeError: _ARRAY_API is not PyCapsule object
ImportError: numpy.core.multiarray failed to import

The above exception was the direct cause of the following exception:

SystemError: <built-in function __import__> returned a result with an error set
ImportError: numpy.core.multiarray failed to import

The above exception was the direct cause of the following exception:

SystemError: <built-in function __import__> returned a result with an error set
ImportError: numpy.core.multiarray failed to import

The above exception was the direct cause of the following exception:

SystemError: <built-in function __import__> returned a result with an error set
ImportError: numpy.core.multiarray failed to import

The above exception was the direct cause of the following exception:

SystemError: <built-in function __import__> returned a result with an error set
ImportError: numpy.core.multiarray failed to import

The above exception was the direct cause of the following exception:

SystemError: <built-in function __import__> returned a result with an error set
ImportError: numpy.core.multiarray failed to import

The above exception was the direct cause of the following exception:

SystemError: <built-in function __import__> returned a result with an error set
capi_return is NULL
Call-back cb_f_in_zvode__user__routines failed.
Traceback (most recent call last):
  File "tools/verify_wheels.py", line 477, in <module>
    results = backend_sim.run(qobj, system_model).result()
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/aer/aerjob.py", line 41, in _wrapper
    return func(self, *args, **kwargs)
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/aer/aerjob.py", line 94, in result
    return self._future.result(timeout=timeout)
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/usr/lib64/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/usr/lib64/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/aer/backends/pulse_simulator.py", line 145, in _run_job
    results = opsolve(openpulse_system)
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/aer/pulse/solver/opsolve.py", line 62, in opsolve
    out = montecarlo.run()
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/aer/pulse/solver/opsolve.py", line 119, in run
    exp_results = parallel_map(unitary_evolution,
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/tools/parallel.py", line 105, in parallel_map
    return [task(values[0], *task_args, **task_kwargs)]
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/qiskit/providers/aer/pulse/solver/unitary.py", line 91, in unitary_evolution
    ODE.integrate(time, step=0)
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/scipy/integrate/_ode.py", line 430, in integrate
    self._y, self.t = mth(self.f, self.jac or (lambda: None),
  File "/home/mtreinish/git/qiskit/qiskit-aer/test-script/lib/python3.8/site-packages/scipy/integrate/_ode.py", line 1004, in run
    y1, t, istate = self.runner(*args)
  File "numeric_integrator_wrapper.pyx", line 30, in numeric_integrator_wrapper.td_ode_rhs_static
ImportError: numpy.core.multiarray failed to import
```

### Steps to reproduce the problem

```
virtualenv aer-test
aer-test/bin/pip install -U qiskit-terra qiskit-aer
aer-test/bin/python tools/verify_wheels.py
```

### What is the expected behavior?
The script works, it's there to sanity check releases.

### Suggested solutions
Fix the issue and release a 0.4.1 package with it fixed. Additionally, we should switch our compile jobs to both run verify_wheels.py and build in the manylinux docker image to prevent this kind of issue in the future.

**Comment by atilag** [comment]
Ok, this seems like a problem with different versions of Numpy being installed/selected at runtime, now that we can reproduce it consistently will fix it soon.

**Comment by atilag** [comment]
Fixed by #609
